### PR TITLE
Replace usage of WenQuanYi Zen Hei by Noto Sans CJK

### PR DESCRIPTION
**Pull request by Sigma-Verma**
## PR Summary

This PR  replaces usage of WenQuanYi Zen Hei with the Noto Sans CJK font, which is already used in other examples and still available on Homebrew. This provides a more convenient and consistent experience for macOS users. The changes were made to the relevant examples in the docs/galleries, and the documentation was updated to reflect the font change.

## PR Checklist

**Documentation and Tests**
- [x] Has pytest style unit tests (and `pytest` passes)
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] New plotting related features are documented with examples.

### CODE REVIEW

File: doc/devel/dependencies.rst [hunk 1]
```diff
@@ -252,7 +252,7 @@ testing the following will be used if they are installed.
 - pytest-xvfb_ to run tests without windows popping up (Linux)
 - pytz_ used to test pytz int
 - sphinx_ used to test our sphinx extensions
-- WenQuanYi Zen Hei and `Noto Sans CJK <https://fonts.google.com/noto/use>`_
+- `Noto Sans CJK <https://fonts.google.com/noto/use>`_
```
Comment (tacaswell) [inline doc/devel/dependencies.rst:255]: I think we still use WenQuanYi Zen Hei in the test suite.

Comment (tacaswell) [reply doc/devel/dependencies.rst:255]: I think this line should be un-changed, but we need to add noto sans to the docs dependency in the next section.

File: doc/users/prev_whats_new/whats_new_3.6.0.rst [hunk 2]
```diff
@@ -558,10 +558,10 @@ them in order to locate a required glyph.
    fig = plt.figure(figsize=(4.75, 1.85))
 
    text = "There are 几个汉字 in between!"
-   fig.text(0.05, 0.85, text, family=["WenQuanYi Zen Hei"])
+   fig.text(0.05, 0.85, text, family=["Noto Sans CJK"])
```
Comment (tacaswell) [inline doc/users/prev_whats_new/whats_new_3.6.0.rst:561]: I suspect we should just remove the to lines that used WQYZH ?

File: galleries/users_explain/text/fonts.py [hunk 3]
```diff
@@ -185,7 +185,7 @@
    fig, ax = plt.subplots()
    ax.text(
        .5, .5, "There are 几个汉字 in between!",
-       family=['DejaVu Sans', 'WenQuanYi Zen Hei'],
+       family=['DejaVu Sans', 'Noto Sans CJK'],
```
Comment (QuLogic) [inline galleries/users_explain/text/fonts.py:188]: This failed to build:
```suggestion
       family=['DejaVu Sans', 'Noto Sans CJK JP'],
```

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
Thanks @Sigma-Verma! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Pull request merged**